Services/CO.CDP.Forms.WebApi/UseCase/UpdateFormSectionAnswersUseCase.cs

```diff
@@ -30,12 +30,30 @@ public async Task<bool> Execute((Guid formId, Guid sectionId, Guid answerSetId,
         }
         else if (sharedConsent.SubmissionState == Persistence.SubmissionState.Submitted)
         {
-            formRepository.ClearTracker();
-            sharedConsent.AnswerSets = sharedConsent.AnswerSets.Where(x => x.Section.Type != Persistence.FormSectionType.Declaration).ToList();
+            var (newSharedConsent, oldToNewIds) = MapSharedConsent(sharedConsent);
 
-            await UpdateOrAddAnswers(formRepository, answerSetId, answers, section, sharedConsent);
-            ClearSharedConsent(sharedConsent);
-            await formRepository.SaveSharedConsentAsync(sharedConsent);
+            if (oldToNewIds.Count > 0)
+            {
+                for (int i = 0; i < answers.Count; i++)
+                {
+                    if (oldToNewIds.ContainsKey(answers[i].Id))
+                    {
+                        answers[i].Id = oldToNewIds[answers[i].Id];
+                    }
+                }
+            }
+
+            if (oldToNewIds.ContainsKey(answerSetId))
+            {
+                await UpdateOrAddAnswers(formRepository, oldToNewIds[answerSetId], answers, section, newSharedConsent);
+            }
+            else
+            {
+                await UpdateOrAddAnswers(formRepository, answerSetId, answers, section, newSharedConsent);
+            }
+
+            formRepository.CheckEntitiesStates(newSharedConsent);
```

Comment:
That's the method mentioned earlier that will help to monitor entity states during debugging. All entities should be of state Unchanged (both all under sharedConsent and newSharedConsent).